Makefile

```diff
@@ -15,27 +15,35 @@ Debug:
 	mkdir Debug
 
 Debug/Makefile: Debug config
-ifndef INSTALL_DIR
-	(cd Debug && cmake .. -DCMAKE_BUILD_TYPE=Debug)
+ifndef INSTALL_PREFIX
+	cd Debug && NUM_DEVS=${NUM_DEVS} DEV_NAMES=${DEV_NAMES} cmake .. -DCMAKE_BUILD_TYPE=Debug
 else
-	(cd Debug && cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX))
+	cd Debug && NUM_DEVS=${NUM_DEVS} DEV_NAMES=${DEV_NAMES} cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX)
 endif
 
 testc: Debug/Makefile
-	(cd Debug && make && $(SUDO) make install)
+ifndef DEVICE
+	cd Debug && make && make test && $(SUDO) make install
+else
+	cd Debug && make && DEVICE=${DEVICE} make test && $(SUDO) make install
+endif
 
 Release:
 	mkdir Release
 
 Release/Makefile: Release config
-ifndef INSTALL_DIR
-	(cd Release && cmake .. -DCMAKE_BUILD_TYPE=Release)
+ifndef INSTALL_PREFIX
+	(cd Release && NUM_DEVS=${NUM_DEVS} DEV_NAMES=${DEV_NAMES} cmake .. -DCMAKE_BUILD_TYPE=Release)
 else
-	(cd Release && cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX))
+	(cd Release && NUM_DEVS=${NUM_DEVS} DEV_NAMES=${DEV_NAMES} cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX))
 endif
 
 relc: Release/Makefile
-	(cd Release && make && $(SUDO) make install)
+ifndef DEVICE
+	cd Release && make && make test && $(SUDO) make install
+else
+	cd Release && make && DEVICE=${DEVICE} make test && $(SUDO) make install
+endif
```

Comment:
I would make a separate test target rather than force the tests to run on each install.